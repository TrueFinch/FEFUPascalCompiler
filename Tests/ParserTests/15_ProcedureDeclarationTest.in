program a;

procedure add1(const a, b: integer; out res: integer);
forward;

procedure add2(const a, b: integer; out res: integer);
begin
    res := a + b;
end;

begin
end.