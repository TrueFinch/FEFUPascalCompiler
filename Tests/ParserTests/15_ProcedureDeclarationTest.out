                                                                            a                                                                             
                                                                            ┼                                                                             
                                                                        MainBlock                                                                         
                              ┌─────────────────────────────────────────────┴────────────────────┬─────────────────────────────────────────────┐          
                           ProcDecl                                                           ProcDecl                                 CompoundStatement  
                          ┌───┴──────────────────────────┐                              ┌────────┴──────────────────────────┐                             
                     ProcHeader                       forward                      ProcHeader                        SubroutineBlock                      
  ┌───────────────┬───────┴──────────────┐                      ┌───────────────┬───────┴──────────────┐                    ┼                             
 add1    FormalParamSection      FormalParamSection            add2    FormalParamSection      FormalParamSection   CompoundStatement                     
         ┌────┬──┬┴─────┐        ┌────┬──┴────┐                        ┌────┬──┬┴─────┐        ┌────┬──┴────┐           ┌───┴                             
       const  a  b  SimpleType  out  res  SimpleType                 const  a  b  SimpleType  out  res  SimpleType     :=                                 
                       ┌┴                    ┌┴                                      ┌┴                    ┌┴        ┌──┴─┐                               
                    integer               integer                                 integer               integer     res   +                               
                                                                                                                         ┌┴─┐                             
                                                                                                                         a  b                             
